doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -614,148 +619,101 @@ syntax: :n:`let fix @ident @binders := @term in` stands for
+on proof mode.
 
-The different kinds of command are described hereafter. They all suppose
-that the terms occurring in the sentences are well-typed.
+By convention, :n:`@command` names begin with uppercase letters, while :n:`@tactic` names begin with lowercase letters.
+
+.. todo:: What about removing the proof part of this grammar from this chapter
+          and putting it somewhere where top-level tactics can be described as well?
+          See also #7583.
 
 .. _gallina-assumptions:
 
 Assumptions
 -----------
 
 Assumptions extend the environment with axioms, parameters, hypotheses
-or variables. An assumption binds an :token:`ident` to a :token:`type`. It is accepted
-by Coq if and only if this :token:`type` is a correct type in the environment
-preexisting the declaration and if :token:`ident` was not previously defined in
-the same module. This :token:`type` is considered to be the type (or
-specification, or statement) assumed by :token:`ident` and we say that :token:`ident`
-has type :token:`type`.
+or variables. An assumption binds an :n:`@ident` to a :n:`@type`. It is accepted
+by Coq if and only if this :n:`@type` is a correct type in the environment
+preexisting the declaration and if :n:`@ident` was not previously defined in
+the same module. This :n:`@type` is considered to be the type (or
+specification, or statement) assumed by :n:`@ident` and we say that :n:`@ident`
+has type :n:`@type`.
 
 .. _Axiom:
 
-.. cmd:: Parameter @ident : @type
-
-   This command links :token:`type` to the name :token:`ident` as its specification in
-   the global context. The fact asserted by :token:`type` is thus assumed as a
-   postulate.
-
-   .. exn:: @ident already exists.
-      :name: @ident already exists. (Axiom)
-      :undocumented:
-
-   .. cmdv:: Parameter {+ @ident } : @type
-
-      Adds several parameters with specification :token:`type`.
-
-   .. cmdv:: Parameter {+ ( {+ @ident } : @type ) }
-
-      Adds blocks of parameters with different specifications.
+.. cmd:: @assumption_token {? Inline {? ( @num ) } } {| {+ ( @assumpt ) } | @assumpt }
+   :name: Axiom; Axioms; Conjecture; Conjectures; Hypothesis; Hypotheses; Parameter; Parameters; Variable; Variables
 
-   .. cmdv:: Local Parameter {+ ( {+ @ident } : @type ) }
-      :name: Local Parameter
+   .. insertprodn assumption_token of_type
 
-      Such parameters are never made accessible through their unqualified name by
-      :cmd:`Import` and its variants. You have to explicitly give their fully
-      qualified name to refer to them.
+   .. prodn::
+      assumption_token ::= {| Axiom | Axioms }
+      | {| Conjecture | Conjectures }
+      | {| Parameter | Parameters }
+      | {| Hypothesis | Hypotheses }
+      | {| Variable | Variables }
+      assumpt ::= {+ @ident_decl } @of_type
+      ident_decl ::= @ident {? @univ_decl }
+      of_type ::= {| : | :> | :>> } @term
```

Comment:
Can we use `type` here? Like so:

```suggestion
      of_type ::= {| : | :> | :>> } @type
```

Reply:
And add a production `type ::= term` in the Terms/Types subsection?

The definition there is not too illuminating:

```
Coq terms are typed. Coq types are recognized by the same syntactic class as term.
We denote by type the semantic subclass of types inside the syntactic class term.
```

Reply:
You find this definition confusing?

Reply:
Yes, not very clear at all.  The wording is stilted.  Is this right?
- Every type is a term.
- Not every term is a type??
- Every term has an associated type, but some terms may have the same type (e.g. (S O) and (O) are both of type nat?  If so, how do you get/compute the type of a term?

Reply:
Everything that you wrote is correct. The typing rules are given in the CIC chapter. The refactoring should make them more accessible.